frontend/src/lib/api.ts

```diff
@@ -2119,10 +2096,33 @@ const api = {
     },
 }
 
-function reportError(method: string, url: string, response: Response, startTime: number): void {
-    const duration = new Date().getTime() - startTime
-    const pathname = new URL(url, location.origin).pathname
-    posthog.capture('client_request_failure', { pathname, method, duration, status: response.status })
+async function handleFetch(url: string, method: string, fetcher: () => Promise<Response>): Promise<Response> {
+    const startTime = new Date().getTime()
+
+    let response
+    let error
+    try {
+        response = await fetcher()
+    } catch (e) {
+        error = e
+    }
+
+    apiStatusLogic.findMounted()?.actions.onApiResponse(response, error)
```

Comment:
I suppose this should `always` be mounted? In that case might be worth doing `useMountedLogic(apiStatusLogic)` in the root `App` component, instead of tying to `sceneLogic`